pkg/api/steve/supportconfigs/generator_test.go

```diff
@@ -0,0 +1,182 @@
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: true,
+			authorized:              false,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusForbidden,
+		},
+		{
+			name:                    "test user requests PAYG, authorized to get PAYG, PAYG not installed (501)",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: true,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusNotImplemented,
+		},
+		{
+			name:                    "test user requests PAYG, not authorized to get PAYG, auth denied (403)",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: true,
+			generateAuthorizedError: false,
+			authorized:              false,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusForbidden,
+		},
+		{
+			name:                    "test user requests BYOL, authorized to get BYOL, BYOL not installed (501)",
+			usePAYG:                 false,
+			generateAdapterError:    false,
+			generateAdapterNotFound: true,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusNotImplemented,
+		},
+		{
+			name:                    "test user requests BYOL, not authorized to get BYOL, auth denied (403)",
+			usePAYG:                 false,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              false,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusForbidden,
+		},
+		{
+			name:                    "test user requests BYOL, authorized to get BYOL, we return output (200)",
+			usePAYG:                 false,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusOK,
+		},
+		{
+			name:                    "test user requests PAYG, autorized to get PAYG, we return output (200)",
+			usePAYG:                 true,
+			generateAdapterError:    false,
+			generateAdapterNotFound: false,
+			generateAuthorizedError: false,
+			authorized:              true,
+			marshalledCSPConfig:     "{}",
+			expectedHTTPCode:        http.StatusOK,
+		},
+	}
+	for _, scenario := range scenarios {
+		test := scenario
+		t.Run(test.name, func(t *testing.T) {
+			h := &Handler{
+				adapterUtil: NewFakeChartUtil(test.generateAdapterError, test.generateAdapterNotFound),
+				cspConfig:   NewFakeCSPConfig(test.generateAuthorizedError, test.authorized, test.marshalledCSPConfig),
+			}
+			reqPath := "/v1/generateSUSERancherSupportConfig"
+			if test.usePAYG {
+				reqPath = reqPath + "?usePAYG=true"
```

Comment:
Nit: I think it would also be good to have a test case for `?usePAYG=false`, but it's not totally necessary since the UI does true or not present.

Reply:
This has the same as usePAYG is absent.